include/seqan3/alignment/cigar_conversion/cigar_from_alignment.hpp

```diff
@@ -0,0 +1,236 @@
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <vector>
+
+#include <seqan3/alphabet/cigar/cigar.hpp>
+#include <seqan3/alphabet/gap/gap.hpp>
+#include <seqan3/utility/views/zip.hpp>
+
+namespace seqan3
+{
+
+/*!\brief Helper struct to specialise soft and hard clipping when using seqan3::cigar_from_alignment.
+ * \ingroup cigar_conversion
+ *
+ * A CIGAR string might have hard or soft clipping at the front or back of a it.
+ * E.g. `2H3S100M3S2H`.
+ */
+struct cigar_clipped_bases
+{
+    uint32_t hard_front{}; //!< The number of hard clipped bases at the front of the CIGAR string.
+    uint32_t hard_back{};  //!< The number of hard clipped bases at the back of the CIGAR string.
+    uint32_t soft_front{}; //!< The number of soft clipped bases at the front of the CIGAR string.
+    uint32_t soft_back{};  //!< The number of soft clipped bases at the back of the CIGAR string.
+};
+
+/*!\brief Creates a cigar string (SAM format) given a seqan3::detail::pairwise_alignment represented by two
+ *        seqan3::aligned_sequence's.
+ * \ingroup cigar_conversion
+ *
+ * \tparam alignment_type  Must model seqan3::detail::pairwise_alignment.
+ * \param  alignment       The alignment, represented by a pair of aligned sequences,
+ *                         to be transformed into cigar vector based on the
+ *                         second (query) sequence.
+ * \param  clipped_bases Gives information on whether the query sequence was cropped (hard clipping) before the
+ *                       alignment or whether part of the query sequence does not take part (soft clipping) in the
+ *                       alignment.
+ * \param  extended_cigar  Whether to print the extended cigar alphabet or not. See cigar operation.
+ * \returns A std::vector\<seqan3::cigar\> representing the alignment.
+ *
+ * \details
+ *
+ * \note The resulting cigar_vector is based on the query sequence, which is the second sequence in the `alignment`
+ *       pair.
+ *
+ * ### Example:
+ *
+ * Given the following alignment reference sequence on top and the query sequence at
+ * the bottom:
+ * ```
+ * ATGG--CGTAGAGC
+ * |||X  |||X|  |
+ * ATGCCCCGTTG--C
+ * ```
+ * In this case, the function seqan3::detail::get_cigar_vector will return the following cigar vector:
+ *
+ * `[('M',4),('I',2),('M',5),('D',2),('M',1)]`.
+ *
+ * The extended cigar string would look like this:
+ *
+ * `[('=',3)('X',1)('I',2)('=',3)('X',1)('=',1)('D',2)('=',1)]`.
+ *
+ * \include test/snippet/alignment/cigar_conversion/cigar_from_alignment.cpp
+ *
+ * ### Soft and Hard clipping
+ *
+ * The terms soft and hard clipping were introduced by the
+ * [SAM specifications](https://samtools.github.io/hts-specs/SAMv1.pdf). A SAM file is only storing a semi alignment
+ * represented by the CIGAR string. the semi alignment of a query sequence is most often the result of a
```

Comment:
Could you add a sentence what a semi alignment is?
```suggestion
 * [SAM specifications](https://samtools.github.io/hts-specs/SAMv1.pdf). A SAM file is only storing a semi alignment
 * represented by the CIGAR string. The semi alignment of a query sequence is most often the result of a
```